#!/usr/bin/env python
PACKAGE = "pid"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
K_scale_enum = gen.enum([ gen.const("scale_tenth", double_t, 0.1, "Scale by 0.1"),
                           gen.const("scale_unity",   double_t, 1.0, "No scaling"),
                           gen.const("scale_ten",  double_t, 10.0, "Scale by 10"),
                           gen.const("scale_hundred", double_t, 100.0, "Scale by 100")],
                           "Scale factor for K setting")

gen.add("Kp_scale", double_t, 0, "Kp scale", 10, 0.1, 100, edit_method = K_scale_enum)
gen.add("Kp", double_t, 0, "Kp", 0.1, -1, 1)
gen.add("Ki_scale", double_t, 0, "Ki scale", 10, 0.1, 100, edit_method = K_scale_enum)
gen.add("Ki", double_t, 0, "Ki", 0.1, -1, 1)
gen.add("Kd_scale", double_t, 0, "Kd scale", 10, 0.1, 100, edit_method = K_scale_enum)
gen.add("Kd", double_t, 0, "Kd", 0.1, -1, 1)

exit(gen.generate(PACKAGE, "pid", "Pid"))
